src/zenml/utils/pipeline_docker_image_builder.py

```diff
@@ -404,6 +381,59 @@ def build_and_push_docker_image(
                     pull=pull_parent_image,
                 )
 
+    def build_and_push_docker_image(
+        self,
+        pipeline_name: str,
+        docker_configuration: DockerConfiguration,
+        stack: "Stack",
+        runtime_configuration: "RuntimeConfiguration",
+        entrypoint: Optional[str] = None,
+    ) -> str:
+        """Builds and pushes a Docker image to run a pipeline.
+
+        Use the image name returned by this method whenever you need to uniquely
+        reference the pushed image in order to pull or run it.
+
+        Args:
+            pipeline_name: Name of the pipeline for which the image(s) should
+                be built.
+            docker_configuration: Docker configuration that specifies what
+                should be included in the image.
+            stack: The stack on which the pipeline will be executed.
+            runtime_configuration: The runtime configuration for the pipeline
+                run.
+            entrypoint: Entrypoint to use for the final image. If left empty,
+                no entrypoint will be included in the image.
+
+        Returns:
+            The Docker repository digest of the pushed image.
+
+        Raises:
+            RuntimeError: If the stack doesn't contain a container registry.
+        """
+        container_registry = stack.container_registry
+        if not container_registry:
+            raise RuntimeError(
+                "Unable to build and push Docker image because stack "
+                f"`{stack.name}` has no container registry."
+            )
+
+        logger.info(
```

Comment:
Wouldn't we want this in `build_docker_image()`?

Reply:
Absolutely, I move it 👍